src/app/page/Home/VolunteerHome.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { makeStyles } from "@material-ui/core/styles";
+import { AppBar, Tabs, Tab, Typography, Box } from "@material-ui/core";
+import {
+  getAllAssignedMissions,
+  getAllStartedMissions,
+  getAllSuggestedMissions,
+  getAllCompletedMissions,
+} from "./missionHelpers";
+import VolunteerHomeMissionList from "./VolunteerHomeMissionList";
+import User from "../../model/User";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+    flexGrow: 1,
+    backgroundColor: theme.palette.background.paper,
+  },
+}));
+
+export default function VolunteerHome({ currentUser }) {
+  const classes = useStyles();
+  const [value, setValue] = React.useState(0);
+  const [assignedMissions, updateAssignedMissions] = useState([]);
+  const [startedMisssions, updateStartedMissions] = useState([]);
+  const [suggestedMissions, updateSuggestedMissions] = useState([]);
+  const [completedMisssions, updateCompletedMissions] = useState([]);
+
+  const fetchAllAssociatedMissions = async () => {
+    const missions = await User.getAllAssociatedMissions(currentUser.uid);
+
+    updateAssignedMissions(getAllAssignedMissions(missions, currentUser));
+    updateStartedMissions(getAllStartedMissions(missions, currentUser));
+    updateSuggestedMissions(getAllSuggestedMissions(missions, currentUser));
+    updateCompletedMissions(getAllCompletedMissions(missions, currentUser));
+  };
+
+  useEffect(() => {
+    fetchAllAssociatedMissions();
+  }, []);
+  const handleChange = (event, newValue) => {
+    setValue(newValue);
+  };
+
+  const handleStartMissionFromAssigned = (missionId) => {
+    User.startMission(currentUser.uid, missionId);
+    const mission = assignedMissions.filter((m) => m.id === missionId);
+    updateStartedMissions(startedMisssions.concat(mission));
+    updateAssignedMissions(assignedMissions.filter((m) => m.id !== missionId));
+  };
+
+  const handleStartMissionFromSuggested = (missionId) => {
+    User.startMission(currentUser.uid, missionId);
+    const mission = suggestedMissions.filter((m) => m.id === missionId);
+    updateStartedMissions(startedMisssions.concat(mission));
+    updateSuggestedMissions(suggestedMissions.filter((m) => m.id !== missionId));
+  };
+
+  const handleDeliveringMissionsFromStarted = (missionId) => {
```

Comment:
These could probably all be consolidated into the same function somehow I'm sure. Probably for another time

Reply:
the price you pay for not using hooks hahaha